### ISSUE TITLE
tableau_parameters.py error whe Parameter name includes the text "Parameter"

### ISSUE THREAD

**Issue by awithey-sedex** [bryantbhowell/tableau_tools#44]
Hi,

The TableauParameters init() code chokes on parameters with the text "Parameter" in the name.

The code assumes they have the format: `[Parameter #]` and tries to cast the second "word" in the parameter name to an int.

This fails if you have a parameter called (for example) "Country Name Parameter" (the xml name attribute is `[Country Name Parameter]`).

One method to fix the error might be to use a regular expression matcher rather than string.find().
e.g.:
```
...

import re
...

        else:
            self.log(u'Parameter XML passed in, finding essential characteristics')
            self.ds_xml = datasource_xml
            params_xml = self.ds_xml.findall(u'./column')
            numberedParameterRegex = re.compile('\[Parameter (\d+)\]')
            for column in params_xml:
                alias = column.get(u'caption')
                internal_name = column.get(u'name')  # type: unicode
                # Parameters are all given internal name [Parameter #], unless they are copies where they
                # end with (copy) h/t Jeff James for discovering
                regexMatch = numberedParameterRegex.match(internal_name)
                if regexMatch and regexMatch.group(1):
                    param_num = int(regexMatch.group(1))
                    # Move up the highest_param_num counter for when you add new ones
                    if param_num > self._highest_param_num:
                        self._highest_param_num = param_num
...
```

Regards,

Andrew

**Comment by bryanthowell-tableau** [comment]
Hi Andrew,
Thank you for providing this solution. I'm implemented it in the code for the 4.6.2 release and will close this when it is fully released

**Comment by bryanthowell-tableau** [comment]
Version 4.6.2 is now released with a fix for this issue. Please reopen if the issue persists.
